src/core.js

```diff
@@ -230,19 +230,23 @@ function resumeProcessing( test ) {
 	// A slight delay to allow this iteration of the event loop to finish (more assertions, etc.)
 	if ( defined.setTimeout ) {
 		setTimeout( function() {
-			var current = test || config.current;
-			if ( current && ( current.semaphore > 0 || current.resumed ) ) {
+			var current;
+
+			// Ensure we are resuming the current test
+			if ( test && test === config.current ) {
+				current = test;
```

Comment:
If I understand correctly, the problem is that we were using `resumed` to mark progression from one asynchronous phase to another (when `semaphore` dips to zero), but treating it as a _test_-level property. So rather than replacing the `test` passed in to `resumeProcessing` here, couldn't we either make `resumed` phase-aware, or explicitly clear it in the synchronized callbacks?

Then again, this feels like a band-aid on that synchronization anyway, so meh. We really should get to the deeper issue within the next couple of months.